operator/pkg/component/component.go

```diff
@@ -719,6 +719,17 @@ func isCoreComponentEnabled(c *CommonComponentFields) bool {
 	return enabled
 }
 
+func getK8sResourceSpec(c *CommonComponentFields) *v1alpha1.KubernetesResourcesSpec {
+	switch t := c.spec.(type) {
+	case *v1alpha1.GatewaySpec:
+		return t.K8S
+	case *v1alpha1.ComponentSpec:
```

Comment:
what is the difference between these two?

Reply:
nothing much anymore but these are just separate types to support any specializations in the future. see istio/api